hazelcast/src/main/java/com/hazelcast/jet/impl/jobupload/JobUploadStatus.java

```diff
@@ -0,0 +1,279 @@
+
+    private final Clock clock;
+
+    // Last processed value
+    private int currentPart;
+
+    // Last processed value
+    private int totalPart;
+
+    // Metadata about the upload
+    private final JobMetaDataParameterObject jobMetaDataParameterObject;
+
+    public JobUploadStatus(JobMetaDataParameterObject parameterObject) {
+        this(parameterObject, Clock.systemUTC());
+    }
+
+    // Get an external Clock for easy unit testing
+    protected JobUploadStatus(JobMetaDataParameterObject parameterObject, Clock clock) {
+        this.jobMetaDataParameterObject = parameterObject;
+        this.clock = clock;
+        changeLastUpdatedTime();
+    }
+
+    /**
+     * Returns if this instance is considered expired or not
+     *
+     * @return true if expired, false otherwise
+     */
+    public boolean isExpired() {
+        Instant now = clock.instant();
+        Duration between = Duration.between(lastUpdatedTime, now);
+        // Compare elapsed time to EXPIRATION_MINUTES
+        long minutes = between.toMinutes();
+        return minutes >= EXPIRATION_MINUTES;
+    }
+
+    /**
+     * Called when this object is removed from JobUploadStore
+     */
+    public void onRemove() {
+        try {
+            if (jobMetaDataParameterObject.getJarPath() != null) {
+                Files.delete(jobMetaDataParameterObject.getJarPath());
+            }
+        } catch (IOException exception) {
+            LOGGER.severe("Error while deleting file : " + jobMetaDataParameterObject.getJarPath(), exception);
+        }
+    }
+
+    /**
+     * Process the part message for the job upload
+     *
+     * @param parameterObject specified message from client
+     * @return JobMetaDataParameterObject if upload is complete
+     * @throws IOException              in case of I/O error
+     * @throws NoSuchAlgorithmException in case of message digest error
+     */
+    public JobMetaDataParameterObject processJobMultipart(JobMultiPartParameterObject parameterObject)
+            throws IOException, NoSuchAlgorithmException {
+
+        validateReceivedParameters(parameterObject);
+
+        validateReceivedPartNumbersAreExpected(parameterObject);
+
+        validatePartChecksum(parameterObject);
+
+        // Parts numbers are good. Save them
+        currentPart = parameterObject.getCurrentPartNumber();
+        totalPart = parameterObject.getTotalPartNumber();
+
+        // If the first part
+        if (currentPart == 1) {
+            createNewTemporaryFile();
+        }
+
+        Path jarPath = jobMetaDataParameterObject.getJarPath();
+
+        // Append data to file
+        try (FileOutputStream outputStream = new FileOutputStream(jarPath.toFile(), true)) {
+            outputStream.write(parameterObject.getPartData(), 0, parameterObject.getPartSize());
+        }
+
+        String message = String.format("Session : %s total file size %d", parameterObject.getSessionId(), Files.size(jarPath));
```

Comment:
wrap it please with `if (LOGGER.isInfoEnabled)` -> we don't want to calculate size if it's not necessary

Reply:
Ok